app/components/learner-group/header.js

```diff
@@ -2,24 +2,50 @@ import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
 import { dropTask } from 'ember-concurrency';
-import { use } from 'ember-could-get-used-to-this';
 import { validatable, Length, NotBlank } from 'ilios-common/decorators/validation';
-import ResolveAsyncValue from 'ilios-common/classes/resolve-async-value';
+import { TrackedAsyncData } from 'ember-async-data';
+import { cached } from '@glimmer/tracking';
 
 @validatable
 export default class LearnerGroupHeaderComponent extends Component {
   @tracked @NotBlank() @Length(3, 60) title;
-  @use cohort = new ResolveAsyncValue(() => [this.args.learnerGroup.cohort]);
-  @use programYear = new ResolveAsyncValue(() => [this.cohort?.programYear]);
-  @use program = new ResolveAsyncValue(() => [this.programYear?.program]);
-  @use school = new ResolveAsyncValue(() => [this.program?.school]);
+
+  @cached
+  get upstreamRelationshipsData() {
+    return new TrackedAsyncData(this.resolveUpstreamRelationships(this.args.learnerGroup));
+  }
+
+  get upstreamRelationships() {
+    return this.upstreamRelationshipsData.isResolved ? this.upstreamRelationshipsData.value : null;
+  }
+
+  get programYear() {
+    return this.upstreamRelationships?.programYear;
+  }
+
+  get program() {
+    return this.upstreamRelationships?.program;
+  }
+
+  get school() {
+    return this.upstreamRelationships?.school;
+  }
 
   get usersOnlyAtThisLevel() {
     return this.args.learnerGroup.usersOnlyAtThisLevel
       ? this.args.learnerGroup.usersOnlyAtThisLevel
       : [];
   }
 
+  async resolveUpstreamRelationships(learnerGroup) {
```

Comment:
:fire:

Reply:
I like this one too! Especially in some of our models where we're loading a big chain, I think this might be useful. It's a potential foot gun though because if any of those connections change this won't re-render. Safe here because we don't ever move a learner group into another cohort or a cohort into another program etc...